Timeout creating VPN routes

After renewing the server certificate on `prod`, deploying `prod.gitlab` resulted in resource creation timeout.

(heavily abbreviated, removed all `Still destroying` and `Still creating` messages):

```
$ make apply

…

Terraform will perform the following actions:

  # aws_ec2_client_vpn_authorization_rule.gitlab must be replaced
-/+ resource "aws_ec2_client_vpn_authorization_rule" "gitlab" {
      ~ client_vpn_endpoint_id = "cvpn-endpoint-06d58661e78958388" -> (known after apply) # forces replacement
      ~ id                     = "cvpn-endpoint-06d58661e78958388,0.0.0.0/0" -> (known after apply)
        # (4 unchanged attributes hidden)
    }

  # aws_ec2_client_vpn_endpoint.gitlab must be replaced
-/+ resource "aws_ec2_client_vpn_endpoint" "gitlab" {
      ~ arn                     = "arn:aws:ec2:us-east-1:542754589326:client-vpn-endpoint/cvpn-endpoint-06d58661e78958388" -> (known after apply)
      ~ dns_name                = "*.cvpn-endpoint-06d58661e78958388.prod.clientvpn.us-east-1.amazonaws.com" -> (known after apply)
      ~ id                      = "cvpn-endpoint-06d58661e78958388" -> (known after apply)
      + self_service_portal_url = (known after apply)
      ~ server_certificate_arn  = "arn:aws:acm:us-east-1:542754589326:certificate/ff65ae38-9f39-48c6-a770-540ab8b06ec3" -> "arn:aws:acm:us-east-1:542754589326:certificate/ac80ba13-2cf8-4b63-8317-aaf4cd8ef996"
        tags                    = {
            "Name"                = "azul-gitlab"
            "billing"             = "hca"
            "component"           = "azul-gitlab"
            "deployment"          = "prod"
            "owner"               = "hannes@ucsc.edu"
            "service"             = "azul"
            "terraform_component" = "gitlab"
        }
        # (11 unchanged attributes hidden)

      - authentication_options { # forces replacement
          - root_certificate_chain_arn     = "arn:aws:acm:us-east-1:542754589326:certificate/ff65ae38-9f39-48c6-a770-540ab8b06ec3" -> null
          - type                           = "certificate-authentication" -> null
            # (3 unchanged attributes hidden)
        }
      + authentication_options { # forces replacement
          + root_certificate_chain_arn     = "arn:aws:acm:us-east-1:542754589326:certificate/ac80ba13-2cf8-4b63-8317-aaf4cd8ef996"
          + type                           = "certificate-authentication"
            # (3 unchanged attributes hidden)
        }

      ~ client_connect_options (known after apply)
      - client_connect_options {
          - enabled             = false -> null
            # (1 unchanged attribute hidden)
        }

      ~ client_login_banner_options (known after apply)
      - client_login_banner_options {
          - enabled     = false -> null
            # (1 unchanged attribute hidden)
        }

      ~ connection_log_options {
          ~ cloudwatch_log_stream = "cvpn-endpoint-06d58661e78958388-us-east-1-2024/06/11-3UOJTs0tMxII" -> (known after apply)
            # (2 unchanged attributes hidden)
        }
    }

  # aws_ec2_client_vpn_network_association.gitlab_0 must be replaced
-/+ resource "aws_ec2_client_vpn_network_association" "gitlab_0" {
      ~ association_id         = "cvpn-assoc-084b9dee50ce645bc" -> (known after apply)
      ~ client_vpn_endpoint_id = "cvpn-endpoint-06d58661e78958388" -> (known after apply) # forces replacement
      ~ id                     = "cvpn-assoc-084b9dee50ce645bc" -> (known after apply)
      ~ vpc_id                 = "vpc-08f117d77ac08342a" -> (known after apply)
        # (1 unchanged attribute hidden)
    }

  # aws_ec2_client_vpn_network_association.gitlab_1 must be replaced
-/+ resource "aws_ec2_client_vpn_network_association" "gitlab_1" {
      ~ association_id         = "cvpn-assoc-0e1ab8f5c14a813e8" -> (known after apply)
      ~ client_vpn_endpoint_id = "cvpn-endpoint-06d58661e78958388" -> (known after apply) # forces replacement
      ~ id                     = "cvpn-assoc-0e1ab8f5c14a813e8" -> (known after apply)
      ~ vpc_id                 = "vpc-08f117d77ac08342a" -> (known after apply)
        # (1 unchanged attribute hidden)
    }

  # aws_ec2_client_vpn_route.gitlab_0 must be replaced
-/+ resource "aws_ec2_client_vpn_route" "gitlab_0" {
      ~ client_vpn_endpoint_id = "cvpn-endpoint-06d58661e78958388" -> (known after apply) # forces replacement
      ~ id                     = "cvpn-endpoint-06d58661e78958388,subnet-0684c75a115306ba9,0.0.0.0/0" -> (known after apply)
      ~ origin                 = "add-route" -> (known after apply)
      ~ type                   = "Nat" -> (known after apply)
        # (3 unchanged attributes hidden)
    }

  # aws_ec2_client_vpn_route.gitlab_1 must be replaced
-/+ resource "aws_ec2_client_vpn_route" "gitlab_1" {
      ~ client_vpn_endpoint_id = "cvpn-endpoint-06d58661e78958388" -> (known after apply) # forces replacement
      ~ id                     = "cvpn-endpoint-06d58661e78958388,subnet-086a7d085dffe194f,0.0.0.0/0" -> (known after apply)
      ~ origin                 = "add-route" -> (known after apply)
      ~ type                   = "Nat" -> (known after apply)
        # (3 unchanged attributes hidden)
    }

Plan: 6 to add, 0 to change, 6 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_ec2_client_vpn_authorization_rule.gitlab: Destroying... [id=cvpn-endpoint-06d58661e78958388,0.0.0.0/0]
aws_ec2_client_vpn_route.gitlab_0: Destroying... [id=cvpn-endpoint-06d58661e78958388,subnet-0684c75a115306ba9,0.0.0.0/0]
aws_ec2_client_vpn_network_association.gitlab_0: Destroying... [id=cvpn-assoc-084b9dee50ce645bc]
aws_ec2_client_vpn_route.gitlab_1: Destroying... [id=cvpn-endpoint-06d58661e78958388,subnet-086a7d085dffe194f,0.0.0.0/0]
aws_ec2_client_vpn_network_association.gitlab_1: Destroying... [id=cvpn-assoc-0e1ab8f5c14a813e8]
aws_ec2_client_vpn_route.gitlab_1: Destruction complete after 1s
aws_ec2_client_vpn_route.gitlab_0: Destruction complete after 1s
aws_ec2_client_vpn_network_association.gitlab_0: Destruction complete after 8m7s
aws_ec2_client_vpn_authorization_rule.gitlab: Destruction complete after 8m16s
aws_ec2_client_vpn_network_association.gitlab_1: Destruction complete after 8m17s
aws_ec2_client_vpn_endpoint.gitlab: Destroying... [id=cvpn-endpoint-06d58661e78958388]
aws_ec2_client_vpn_endpoint.gitlab: Destruction complete after 1s
aws_ec2_client_vpn_endpoint.gitlab: Creating...
aws_ec2_client_vpn_endpoint.gitlab: Creation complete after 3s [id=cvpn-endpoint-05cefdac46aecf285]
aws_ec2_client_vpn_network_association.gitlab_0: Creating...
aws_ec2_client_vpn_network_association.gitlab_1: Creating...
aws_ec2_client_vpn_authorization_rule.gitlab: Creating...
aws_ec2_client_vpn_route.gitlab_0: Creating...
aws_ec2_client_vpn_route.gitlab_1: Creating...
aws_ec2_client_vpn_authorization_rule.gitlab: Creation complete after 7m36s [id=cvpn-endpoint-05cefdac46aecf285,0.0.0.0/0]
aws_ec2_client_vpn_network_association.gitlab_1: Creation complete after 7m37s [id=cvpn-assoc-04dbde51f46b6791e]
aws_ec2_client_vpn_network_association.gitlab_0: Creation complete after 7m38s [id=cvpn-assoc-0e8bc9f2015cfcca9]
╷
│ Error: waiting for EC2 Client VPN Route (cvpn-endpoint-05cefdac46aecf285,subnet-0684c75a115306ba9,0.0.0.0/0) create: timeout while waiting for state to become 'active' (last state: 'creating', timeout: 4m0s)
│ 
│   with aws_ec2_client_vpn_route.gitlab_0,
│   on gitlab.tf.json line 1372, in resource[15].aws_ec2_client_vpn_route[0].gitlab_0:
│ 1372:                     }
│ 
╵
╷
│ Error: waiting for EC2 Client VPN Route (cvpn-endpoint-05cefdac46aecf285,subnet-086a7d085dffe194f,0.0.0.0/0) create: timeout while waiting for state to become 'active' (last state: 'creating', timeout: 4m0s)
│ 
│   with aws_ec2_client_vpn_route.gitlab_1,
│   on gitlab.tf.json line 1379, in resource[15].aws_ec2_client_vpn_route[1].gitlab_1:
│ 1379:                     }
│ 
╵
make: *** [apply] Error 1
```

A second attempt resulted in a plan that wanted to replace those two resources.

```
$ make apply
…
Terraform will perform the following actions:

  # aws_ec2_client_vpn_route.gitlab_0 is tainted, so must be replaced
-/+ resource "aws_ec2_client_vpn_route" "gitlab_0" {
      ~ id                     = "cvpn-endpoint-05cefdac46aecf285,subnet-0684c75a115306ba9,0.0.0.0/0" -> (known after apply)
      ~ origin                 = "add-route" -> (known after apply)
      ~ type                   = "Nat" -> (known after apply)
        # (4 unchanged attributes hidden)
    }

  # aws_ec2_client_vpn_route.gitlab_1 is tainted, so must be replaced
-/+ resource "aws_ec2_client_vpn_route" "gitlab_1" {
      ~ id                     = "cvpn-endpoint-05cefdac46aecf285,subnet-086a7d085dffe194f,0.0.0.0/0" -> (known after apply)
      ~ origin                 = "add-route" -> (known after apply)
      ~ type                   = "Nat" -> (known after apply)
        # (4 unchanged attributes hidden)
    }

Plan: 2 to add, 0 to change, 2 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: no

Apply cancelled.
make: *** [apply] Error 1
```

Untainting them …


```
((.venv) ) MacBookPro:gitlab hannes$ terraform untaint aws_ec2_client_vpn_route.gitlab_1
Resource instance aws_ec2_client_vpn_route.gitlab_1 has been successfully untainted.
((.venv) ) MacBookPro:gitlab hannes$ terraform untaint aws_ec2_client_vpn_route.gitlab_0
Resource instance aws_ec2_client_vpn_route.gitlab_0 has been successfully untainted.
```

– and trying a third time resulted in an empty plan:

```
((.venv) ) MacBookPro:gitlab hannes$ make apply
…

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

--- a/terraform/gitlab/gitlab.tf.json.template.py
+++ b/terraform/gitlab/gitlab.tf.json.template.py
@@ -1227,7 +1227,11 @@ emit_tf({} if config.terraform_component != 'gitlab' else {
             f'gitlab_{zone}': {
                 'client_vpn_endpoint_id': '${aws_ec2_client_vpn_endpoint.gitlab.id}',
                 'target_vpc_subnet_id': '${aws_subnet.gitlab_public_%s.id}' % zone,
-                'destination_cidr_block': all_ipv4
+                'destination_cidr_block': all_ipv4,
+                'timeouts': {
+                    # The default is 4 min, which is too short
+                    'create': '10m'
+                }
             }
             for zone in range(num_zones)
             if not split_tunnel

